### ISSUE TITLE
Getting different results using one_hot(...) than when using plain numpy arrays

### ISSUE THREAD

**Issue by thilow** [Microsoft/CNTK#1333]
in the Example [char-rnn](https://github.com/Microsoft/CNTK/blob/master/Examples/Text/CharacterLM/char-rnn.py) the original code uses numpy arrays as one-hot-vectors instead of creating them with `one_hot(...)`.
When I try to use `one_hot` the training output changes (with current master).

I Replaced
`
def get_data(p, minibatch_size, data, char_to_ix, vocab_dim): 
     
     # the character LM predicts the next character so get sequences offset by 1 
     xi = [char_to_ix[ch] for ch in data[p:p+minibatch_size]] 
     yi = [char_to_ix[ch] for ch in data[p+1:p+minibatch_size+1]]

     # a slightly inefficient way to get one-hot vectors but fine for low vocab (like char-lm) 
     X = np.eye(vocab_dim, dtype=np.float32)[xi] 
     Y = np.eye(vocab_dim, dtype=np.float32)[yi]

    # return a list of numpy arrays for each of X (features) and Y (labels)
    return [X], [Y]
`

by 
`
def get_data(p, minibatch_size, data, char_to_ix, vocab_dim):

    # the character LM predicts the next character so get sequences offset by 1
    xi = [int(char_to_ix[ch]) for ch in data[p:p+minibatch_size]]
    yi = [int(char_to_ix[ch]) for ch in data[p+1:p+minibatch_size+1]]

    X = C.one_hot([xi], vocab_dim)
    Y = C.one_hot([yi], vocab_dim)

    # return a list of numpy arrays for each of X (features) and Y (labels)
    return X, Y
`

I expected to see the same training output, but they consistently differ.
Output original version:
 Minibatch[   1- 100]: loss = 3.928362 * 10000, metric = 85.5% * 10000
 Minibatch[ 101- 200]: loss = 3.467598 * 10000, metric = 85.4% * 10000
 Minibatch[ 201- 300]: loss = 3.312312 * 10000, metric = 85.1% * 10000

Output modified version:
 Minibatch[   1- 100]: loss = 3.930405 * 10000, metric = 85.5% * 10000
 Minibatch[ 101- 200]: loss = 3.480757 * 10000, metric = 85.4% * 10000
 Minibatch[ 201- 300]: loss = 3.323780 * 10000, metric = 85.1% * 10000
